core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -163,17 +168,18 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
     * Choose the log to clean next and add it to the in-progress set. We recompute this
     * each time from the full set of logs to allow logs to be dynamically added to the pool of logs
     * the log manager maintains.
+    * Returns a tuple of an Option of the log selected to be cleaned and the reason it was selected.
     */
-  def grabFilthiestCompactedLog(time: Time, preCleanStats: PreCleanStats = new PreCleanStats()): Option[LogToClean] = {
+  def grabFilthiestCompactedLog(time: Time, preCleanStats: PreCleanStats = new PreCleanStats()): (Option[LogToClean], LogCleaningReason) = {
```

Comment:
Could LogCleaningReason be included as a field in LogToClean?

Reply:
I've moved this into the LogToClean class as a var in the helper. We now set this var in `grabFilthiestCompactedLog` and I've expanded on some of the current testing in `LogCleanerManagerTest` to check that the reason is correct when using `grabFilthiestCompactedLog`